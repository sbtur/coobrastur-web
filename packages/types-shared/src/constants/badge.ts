export const BADGE_VARIANTS = {
  default: 'default',
  secondary: 'secondary',
  outline: 'outline',
  success: 'success',
  danger: 'danger',
  warning: 'warning',
  neutral: 'neutral',
} as const;
