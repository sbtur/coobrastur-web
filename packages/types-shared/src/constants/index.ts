export * from './categories';
export * from './badge';
export * from './company-plans';
