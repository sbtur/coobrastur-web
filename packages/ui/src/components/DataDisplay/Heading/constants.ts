const ALIGN = ['left', 'center', 'right'] as const;
const SPACING = ['sm', 'md', 'lg', 'xl'] as const;
const ELEMENT = ['header', 'div'] as const;

export { ALIGN, ELEMENT, SPACING };
