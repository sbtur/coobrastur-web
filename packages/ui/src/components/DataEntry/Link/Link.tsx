import React from 'react';

import { cva, VariantProps } from 'class-variance-authority';

import { cn } from '@workspace/ui/lib/utils';

const linkVariants = cva('mt-2 inline-flex items-center', {
  variants: {
    variant: {
      primary: 'text-primary hover:text-primary-hover',
      secondary: 'text-secondary hover:text-secondary-hover',
      neutral: 'text-white hover:text-gray-400',
    },
    size: {
      small: 'text-sm',
      medium: 'text-base',
      large: 'text-lg',
    },
  },
  defaultVariants: {
    variant: 'primary',
    size: 'medium',
  },
});

export interface LinkProps extends VariantProps<typeof linkVariants> {
  children: React.ReactNode;
  href: string;
  className?: string;
  renderComponent?: (props: LinkProps) => React.ReactElement;
}

export const Link = ({
  children,
  href,
  renderComponent,
  variant = 'primary',
  size,
  ...props
}: LinkProps) => {
  const linkProps = {
    children,
    href,
    className: cn(linkVariants({ variant, size }), props.className),
    ...props,
  };

  if (renderComponent) {
    return renderComponent(linkProps);
  }

  return <a {...linkProps}>{children}</a>;
};
