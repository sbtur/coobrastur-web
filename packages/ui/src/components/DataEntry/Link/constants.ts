export const LINK_VARIANTS = {
  primary: 'primary',
  secondary: 'secondary',
  white: 'white',
  neutral: 'neutral',
} as const;
export type LinkVariant = keyof typeof LINK_VARIANTS;
