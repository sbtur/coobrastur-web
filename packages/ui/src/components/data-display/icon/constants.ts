const ICON_VARIANTS = ['primary', 'secondary', 'neutral', 'white'] as const;

const ICON_SIZES = {
  sm: 14,
  md: 16,
  lg: 20,
  xl: 24,
} as const;

export { ICON_SIZES, ICON_VARIANTS };
