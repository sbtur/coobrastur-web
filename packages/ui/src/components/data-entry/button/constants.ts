export const BUTTON_VARIANTS = {
  default: 'default',
  secondary: 'secondary',
  danger: 'danger',
  outline: 'outline',
  ghost: 'ghost',
  success: 'success',
  warning: 'warning',
  white: 'white',
  'outline-white': 'outline-white',
  link: 'link',
} as const;

export const BUTTON_SIZES = {
  default: 'default',
  sm: 'sm',
  lg: 'lg',
  icon: 'icon',
} as const;

export type ButtonVariant = keyof typeof BUTTON_VARIANTS;
export type ButtonSize = keyof typeof BUTTON_SIZES;
