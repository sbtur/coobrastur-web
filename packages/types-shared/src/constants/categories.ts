import { BadgeVariant } from 'src/types';

export const CATEGORY_LABELS = {
  GOLD: 'Gold',
  SILVER: 'Silver',
  DIAMANTE: 'Diamante',
} as const;

export const CATEGORY_COLORS: Record<
  (typeof CATEGORY_LABELS)[keyof typeof CATEGORY_LABELS],
  BadgeVariant
> = {
  [CATEGORY_LABELS.GOLD]: 'warning',
  [CATEGORY_LABELS.SILVER]: 'neutral',
  [CATEGORY_LABELS.DIAMANTE]: 'default',
};
