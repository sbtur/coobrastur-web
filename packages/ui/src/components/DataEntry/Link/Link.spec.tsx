import { LINK_VARIANTS } from './constants';
import { Link } from './Link';

import { describe, expect, it, render } from '@ui/tests';

const variants = Object.values(LINK_VARIANTS);

describe('Link', () => {
  it.each(variants)('should create a snapshot for %s variant', variant => {
    expect(
      render(
        <Link href="/test" variant={variant}>
          {variant.charAt(0).toUpperCase() + variant.slice(1)}
        </Link>,
      ),
    ).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const { getByText } = render(<Link href="/test">Default</Link>);
    expect(getByText('Default')).toBeInTheDocument();
    expect(getByText('Default')).toHaveAttribute('href', '/test');
  });

  it('should apply target attribute', () => {
    const { getByText } = render(
      <Link href="/test" target="_blank">
        Target Blank
      </Link>,
    );
    expect(getByText('Target Blank')).toHaveAttribute('target', '_blank');
  });
});
