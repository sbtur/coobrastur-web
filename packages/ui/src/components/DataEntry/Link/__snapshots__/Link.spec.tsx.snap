// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > should create a snapshot for neutral variant 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="inline-flex items-center text-highlight hover:text-highlight-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Primary
      </a>
    </div>
    <div>
      <a
        class="inline-flex items-center text-secondary-300 hover:text-secondary-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Secondary
      </a>
    </div>
    <div>
      <a
        class="inline-flex items-center text-white [&_svg]:text-current text-base"
        href="/test"
      >
        White
      </a>
    </div>
    <div>
      <a
        class="inline-flex items-center text-text hover:text-text-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Neutral
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="inline-flex items-center text-text hover:text-text-hover [&_svg]:text-current text-base"
      href="/test"
    >
      Neutral
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Link > should create a snapshot for primary variant 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="inline-flex items-center text-highlight hover:text-highlight-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Primary
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="inline-flex items-center text-highlight hover:text-highlight-hover [&_svg]:text-current text-base"
      href="/test"
    >
      Primary
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Link > should create a snapshot for secondary variant 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="inline-flex items-center text-highlight hover:text-highlight-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Primary
      </a>
    </div>
    <div>
      <a
        class="inline-flex items-center text-secondary-300 hover:text-secondary-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Secondary
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="inline-flex items-center text-secondary-300 hover:text-secondary-hover [&_svg]:text-current text-base"
      href="/test"
    >
      Secondary
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Link > should create a snapshot for white variant 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="inline-flex items-center text-highlight hover:text-highlight-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Primary
      </a>
    </div>
    <div>
      <a
        class="inline-flex items-center text-secondary-300 hover:text-secondary-hover [&_svg]:text-current text-base"
        href="/test"
      >
        Secondary
      </a>
    </div>
    <div>
      <a
        class="inline-flex items-center text-white [&_svg]:text-current text-base"
        href="/test"
      >
        White
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="inline-flex items-center text-white [&_svg]:text-current text-base"
      href="/test"
    >
      White
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
